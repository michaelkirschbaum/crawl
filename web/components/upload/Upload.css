ul {
  list-style-type: none;
  display: flex;
}

li {
  width: fit-content;
  display: inline-block;
  margin-right: 20px;
}

img {
  width: 200px;
  length: 200px;
}
