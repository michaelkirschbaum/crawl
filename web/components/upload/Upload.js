import React, { Component } from 'react'
import "./Upload.css"

class Upload extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      file: null,
      projects: [],
      signedUrl: ''
    }

    this.onChange = this.onChange.bind(this)
    this.onFileChange = this.onFileChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  componentDidMount() {
    // get projects
    fetch('http://localhost:8081/mockups/get')
      .then(res => res.json())
      .then(resJson => {
        this.setState({ projects: resJson })
      })
      .catch(err => console.log(err))
  }

  onFileChange(event) {
    // set file
    this.setState({file: event.target.files[0]})

    // request pre-signed url
    fetch(`http://localhost:8081/mockups/signUrl?fileName=${event.target.files[0]}`)
      .then(res => {
        if (res.ok) {
          return res.json()
        } else {
          throw new error('request failed')
        }
      })
      .then(resJson => {
        this.setState({ signedUrl: resJson.signedRequest })
      })
      .catch(err => console.log(err))
  }

  onChange(event) {
    this.setState({name: event.target.value})
  }

  handleSubmit(event) {
    event.preventDefault()

    // upload image to s3
    fetch(this.state.signedUrl, {
      method: "POST",
      body: this.state.file
    })
    .then(res => {
      if (!res.ok) throw new Error(`${response.status}: ${response.statusText}`)
    })
  }

  render() {
    return (
      <div className="Upload">
        <h1>Projects</h1>

        <ul>
          {this.state.projects.map(project => {
              return <li>{project.name} {project.uri}</li>
          })}
        </ul>

        <form onSubmit={this.handleSubmit}>
          Name: <input type="text" name="name" onChange={this.onChange} />
          <input type="file" accept="image/png, image/jpeg" name="mockup" onChange={this.onFileChange} />
          <input type="submit" value="Submit" />
        </form>
      </div>
    )
  }
}

export default Upload
