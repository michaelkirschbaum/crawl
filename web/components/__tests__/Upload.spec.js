import React from 'react'
import Upload from '../../components/upload/Upload'
import fetchMock from 'fetch-mock'
import { shallow } from 'enzyme'

describe('Upload', () => {
  fetchMock.get('*', JSON.stringify({res: ''}))

  it('should render correctly', () => {
    const component = shallow(<Upload />)
    expect(component).toMatchSnapshot()
  })

  it('should lookup projects', () => {
    expect(false).toBe(true)
  })

  it('should show error when project lookup fails', () => {
    expect(false).toBe(true)
  })

  it('should show spinner when loading project', () => {
    expect(false).toBe(true)
  })

  it('set error when loading project fails', () => {
    expect(false).toBe(true)
  })

  describe('submit is clicked', () => {
    it('should save project', () => {
      expect(false).toBe(true)
    })

    it('should set error when unable to save project', () => {
      expect(false).toBe(true)
    })

    it('should upload image', () => {
      expect(false).toBe(true)
    })

    it('should set project in state', () => {
      expect(false).toBe(true)
    })
  })
})
