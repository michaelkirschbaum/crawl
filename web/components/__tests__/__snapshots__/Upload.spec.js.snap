// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Upload />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <onlyUpdateForKeys(Loader)
          className=""
          color="#000000"
          height={15}
          heightUnit="px"
          loading={false}
          margin="2px"
          radius={2}
          radiusUnit="px"
          width={5}
          widthUnit="px"
        />,
        <h1>
          Projects
        </h1>,
        <ul />,
        <form
          onSubmit={[Function]}
        >
          <input
            name="name"
            onChange={[Function]}
            placeholder="Name"
            type="text"
          />
          <input
            accept="image/png"
            name="mockup"
            onChange={[Function]}
            type="file"
          />
          <input
            type="submit"
            value="Submit"
          />
        </form>,
      ],
      "className": "Upload",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "",
          "color": "#000000",
          "height": 15,
          "heightUnit": "px",
          "loading": false,
          "margin": "2px",
          "radius": 2,
          "radiusUnit": "px",
          "width": 5,
          "widthUnit": "px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Projects",
        },
        "ref": null,
        "rendered": "Projects",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="name"
              onChange={[Function]}
              placeholder="Name"
              type="text"
            />,
            <input
              accept="image/png"
              name="mockup"
              onChange={[Function]}
              type="file"
            />,
            <input
              type="submit"
              value="Submit"
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "name",
              "onChange": [Function],
              "placeholder": "Name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "accept": "image/png",
              "name": "mockup",
              "onChange": [Function],
              "type": "file",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
              "value": "Submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <onlyUpdateForKeys(Loader)
            className=""
            color="#000000"
            height={15}
            heightUnit="px"
            loading={false}
            margin="2px"
            radius={2}
            radiusUnit="px"
            width={5}
            widthUnit="px"
          />,
          <h1>
            Projects
          </h1>,
          <ul />,
          <form
            onSubmit={[Function]}
          >
            <input
              name="name"
              onChange={[Function]}
              placeholder="Name"
              type="text"
            />
            <input
              accept="image/png"
              name="mockup"
              onChange={[Function]}
              type="file"
            />
            <input
              type="submit"
              value="Submit"
            />
          </form>,
        ],
        "className": "Upload",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "color": "#000000",
            "height": 15,
            "heightUnit": "px",
            "loading": false,
            "margin": "2px",
            "radius": 2,
            "radiusUnit": "px",
            "width": 5,
            "widthUnit": "px",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Projects",
          },
          "ref": null,
          "rendered": "Projects",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="name"
                onChange={[Function]}
                placeholder="Name"
                type="text"
              />,
              <input
                accept="image/png"
                name="mockup"
                onChange={[Function]}
                type="file"
              />,
              <input
                type="submit"
                value="Submit"
              />,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "name",
                "onChange": [Function],
                "placeholder": "Name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "accept": "image/png",
                "name": "mockup",
                "onChange": [Function],
                "type": "file",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
                "value": "Submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
